### PR TITLE
RP2040: add GP15

### LINKED ISSUE #4034: Missing GP15 in board module for Raspberry Pi Pico
CircuitPython version: 6.1.0-rc.1-51-gfb1e0106b

dir(board) yield the output of:
['__class__', 'A0', 'A1', 'A2', 'GP0', 'GP1', 'GP10', 'GP11', 'GP12', 'GP13', 'GP14', 'GP16', 'GP17', 'GP18', 'GP19', 'GP2', 'GP20', 'GP21', 'GP22', 'GP25', 'GP26', 'GP26_A0', 'GP27', 'GP27_A1', 'GP28', 'GP28_A2', 'GP3', 'GP4', 'GP5', 'GP6', 'GP7', 'GP8', 'GP9', 'LED']

There are missing GP15 in the list.

### PR DESCRIPTION
**Pull request by dhalbert**
Fixes #4034.

It looks like GP15 should be available after USB reset for general use.

### CODE REVIEW

**Review by ladyada — DISMISSED** [review]

**Review by tannewt — CHANGES_REQUESTED** [review]
You'll need to remove the pin in use bit too. /me looks

**Comment by tannewt** [comment]
https://github.com/adafruit/circuitpython/blob/main/ports/raspberrypi/common-hal/microcontroller/Pin.c#L145-L150

File: ports/raspberrypi/common-hal/microcontroller/Pin.c [hunk 1]
```diff
@@ -63,12 +63,7 @@ void never_reset_pin_number(uint8_t pin_number) {
 }
 
 void reset_pin_number(uint8_t pin_number) {
-    if (pin_number >= 32
-#if TUD_OPT_RP2040_USB_DEVICE_ENUMERATION_FIX
-    // Pin 15 is used for Errata so we don't mess with it.
-        || pin_number == 15
-#endif
-        ) {
+    if (pin_number >= 32) {
```
Comment (dhalbert) [inline ports/raspberrypi/common-hal/microcontroller/Pin.c:66]: Should this be 30?

Comment (dhalbert) [reply ports/raspberrypi/common-hal/microcontroller/Pin.c:66]: And also above on line 58, should that be 30?

**Review by dhalbert — COMMENTED** [review]
@tannewt See my "review" comments for question re two places where top pin is assume to be 32 instead of 30.

**Comment by dhalbert** [comment]
Thanks - I grepped for `GP15`, but not a bare 15.

File: ports/raspberrypi/common-hal/microcontroller/Pin.c (hunk 1 above)
Comment (tannewt) [reply ports/raspberrypi/common-hal/microcontroller/Pin.c:66]: Ya, though it isn't urgent. We don't have objects with numbers higher.

**Review by tannewt — APPROVED** [review]
Thank you!

**Review by jepler — APPROVED** [review]
Looks fine, no testing performed.

**Pull request merged**